manifests/timer_wrapper.pp

```diff
@@ -0,0 +1,92 @@
+# @param command the command for the systemd servie to execute
+# @param user the user to run the command as
+# @param on_active_sec run service relative to the time when the timer was activated
+# @param on_boot_sec run service relative to when the machine was booted
+# @param on_start_up_sec run service relative to when the service manager was started
+# @param on_unit_active_sec run service relative to when the unit was last activated
+# @param on_unit_inactive_sec run service relative to when the unit was last deactivated
+# @param on_calendar the calendar event expressions time to run the service
+# @param service_overrides override for the`[Service]` section of the service
+#   example: `'service_overrides' => { 'Wants' => 'network-online.target' }`
+# @param timer_overrides override for the`[Timer]` section of the timer
+#   example: `'timer_overrides' => { 'OnBootSec' => '10' }`
+# @param service_unit_overrides override for the`[Unit]` section of the service
+#   example: `'service_unit_overrides' => { 'Wants' => 'network-online.target' }`
+# @param timer_unit_overrides override for the `[Unit]` section of the timer
+#   can not be used in combination with `additional_timer_params`
+define systemd::timer_wrapper (
+  Enum['present', 'absent']                 $ensure,
+  Optional[String[1]]                       $command = undef,
+  Optional[String[1]]                       $user = undef,
+  Optional[Systemd::Unit::Timer::TimerSpec] $on_active_sec = undef,
+  Optional[Systemd::Unit::Timer::TimerSpec] $on_boot_sec = undef,
+  Optional[Systemd::Unit::Timer::TimerSpec] $on_start_up_sec = undef,
+  Optional[Systemd::Unit::Timer::TimerSpec] $on_unit_active_sec = undef,
+  Optional[Systemd::Unit::Timer::TimerSpec] $on_unit_inactive_sec = undef,
+  Optional[Systemd::Unit::Timer::TimerSpec] $on_calendar = undef,
+  Systemd::Unit::Service                    $service_overrides         = {},
+  Systemd::Unit::Timer                      $timer_overrides           = {},
+  Systemd::Unit::Unit                       $timer_unit_overrides      = {},
+  Systemd::Unit::Unit                       $service_unit_overrides    = {},
+) {
+  $_timer_spec = {
+    'OnActiveSec'       => $on_active_sec,
+    'OnBootSec'         => $on_boot_sec,
+    'OnStartUpSec'      => $on_start_up_sec,
+    'OnUnitActiveSec'   => $on_unit_active_sec,
+    'OnUnitInactiveSec' => $on_unit_inactive_sec,
+    'OnCalendar'        => $on_calendar,
+  }.filter |$k, $v| { $v =~ NotUndef }
+
+  if $ensure == 'present' {
+    if $_timer_spec == {} {
+      fail('At least one of on_active_sec,
+        on_boot_sec,
+        on_start_up_sec,
+        on_unit_active_sec,
+        on_unit_inactive_sec,
+        or on_calendar must be set'
+      )
+    }
+    if ! $command {
+      fail('command must be set')
+    }
+  }
+
+  $service_ensure = $ensure ? { 'absent' => false,  default  => true, }
+  $unit_name = regsubst($title, '/' , '_', 'G')
```

Comment:
```suggestion
  $unit_name = systemd::escape($title)
```
may as well use the supplied function which covers even stranger cases.

Reply:
Great hint. Implemented, thanks!